<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">

<style type="text/css">
@font-face {
font-family: octicons-link;
src: url(data:font/woff;charset=utf-8;base64,d09GRgABAAAAAAZwABAAAAAACFQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABEU0lHAAAGaAAAAAgAAAAIAAAAAUdTVUIAAAZcAAAACgAAAAoAAQAAT1MvMgAAAyQAAABJAAAAYFYEU3RjbWFwAAADcAAAAEUAAACAAJThvmN2dCAAAATkAAAABAAAAAQAAAAAZnBnbQAAA7gAAACyAAABCUM+8IhnYXNwAAAGTAAAABAAAAAQABoAI2dseWYAAAFsAAABPAAAAZwcEq9taGVhZAAAAsgAAAA0AAAANgh4a91oaGVhAAADCAAAABoAAAAkCA8DRGhtdHgAAAL8AAAADAAAAAwGAACfbG9jYQAAAsAAAAAIAAAACABiATBtYXhwAAACqAAAABgAAAAgAA8ASm5hbWUAAAToAAABQgAAAlXu73sOcG9zdAAABiwAAAAeAAAAME3QpOBwcmVwAAAEbAAAAHYAAAB/aFGpk3jaTY6xa8JAGMW/O62BDi0tJLYQincXEypYIiGJjSgHniQ6umTsUEyLm5BV6NDBP8Tpts6F0v+k/0an2i+itHDw3v2+9+DBKTzsJNnWJNTgHEy4BgG3EMI9DCEDOGEXzDADU5hBKMIgNPZqoD3SilVaXZCER3/I7AtxEJLtzzuZfI+VVkprxTlXShWKb3TBecG11rwoNlmmn1P2WYcJczl32etSpKnziC7lQyWe1smVPy/Lt7Kc+0vWY/gAgIIEqAN9we0pwKXreiMasxvabDQMM4riO+qxM2ogwDGOZTXxwxDiycQIcoYFBLj5K3EIaSctAq2kTYiw+ymhce7vwM9jSqO8JyVd5RH9gyTt2+J/yUmYlIR0s04n6+7Vm1ozezUeLEaUjhaDSuXHwVRgvLJn1tQ7xiuVv/ocTRF42mNgZGBgYGbwZOBiAAFGJBIMAAizAFoAAABiAGIAznjaY2BkYGAA4in8zwXi+W2+MjCzMIDApSwvXzC97Z4Ig8N/BxYGZgcgl52BCSQKAA3jCV8CAABfAAAAAAQAAEB42mNgZGBg4f3vACQZQABIMjKgAmYAKEgBXgAAeNpjYGY6wTiBgZWBg2kmUxoDA4MPhGZMYzBi1AHygVLYQUCaawqDA4PChxhmh/8ODDEsvAwHgMKMIDnGL0x7gJQCAwMAJd4MFwAAAHjaY2BgYGaA4DAGRgYQkAHyGMF8NgYrIM3JIAGVYYDT+AEjAwuDFpBmA9KMDEwMCh9i/v8H8sH0/4dQc1iAmAkALaUKLgAAAHjaTY9LDsIgEIbtgqHUPpDi3gPoBVyRTmTddOmqTXThEXqrob2gQ1FjwpDvfwCBdmdXC5AVKFu3e5MfNFJ29KTQT48Ob9/lqYwOGZxeUelN2U2R6+cArgtCJpauW7UQBqnFkUsjAY/kOU1cP+DAgvxwn1chZDwUbd6CFimGXwzwF6tPbFIcjEl+vvmM/byA48e6tWrKArm4ZJlCbdsrxksL1AwWn/yBSJKpYbq8AXaaTb8AAHja28jAwOC00ZrBeQNDQOWO//sdBBgYGRiYWYAEELEwMTE4uzo5Zzo5b2BxdnFOcALxNjA6b2ByTswC8jYwg0VlNuoCTWAMqNzMzsoK1rEhNqByEyerg5PMJlYuVueETKcd/89uBpnpvIEVomeHLoMsAAe1Id4AAAAAAAB42oWQT07CQBTGv0JBhagk7HQzKxca2sJCE1hDt4QF+9JOS0nbaaYDCQfwCJ7Au3AHj+LO13FMmm6cl7785vven0kBjHCBhfpYuNa5Ph1c0e2Xu3jEvWG7UdPDLZ4N92nOm+EBXuAbHmIMSRMs+4aUEd4Nd3CHD8NdvOLTsA2GL8M9PODbcL+hD7C1xoaHeLJSEao0FEW14ckxC+TU8TxvsY6X0eLPmRhry2WVioLpkrbp84LLQPGI7c6sOiUzpWIWS5GzlSgUzzLBSikOPFTOXqly7rqx0Z1Q5BAIoZBSFihQYQOOBEdkCOgXTOHA07HAGjGWiIjaPZNW13/+lm6S9FT7rLHFJ6fQbkATOG1j2OFMucKJJsxIVfQORl+9Jyda6Sl1dUYhSCm1dyClfoeDve4qMYdLEbfqHf3O/AdDumsjAAB42mNgYoAAZQYjBmyAGYQZmdhL8zLdDEydARfoAqIAAAABAAMABwAKABMAB///AA8AAQAAAAAAAAAAAAAAAAABAAAAAA==) format('woff');
}
body {
-webkit-text-size-adjust: 100%;
text-size-adjust: 100%;
color: #333;
font-family: "Helvetica Neue", Helvetica, "Segoe UI", Arial, freesans, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
font-size: 16px;
line-height: 1.6;
word-wrap: break-word;
}
a {
background-color: transparent;
}
a:active,
a:hover {
outline: 0;
}
strong {
font-weight: bold;
}
h1 {
font-size: 2em;
margin: 0.67em 0;
}
img {
border: 0;
}
hr {
box-sizing: content-box;
height: 0;
}
pre {
overflow: auto;
}
code,
kbd,
pre {
font-family: monospace, monospace;
font-size: 1em;
}
input {
color: inherit;
font: inherit;
margin: 0;
}
html input[disabled] {
cursor: default;
}
input {
line-height: normal;
}
input[type="checkbox"] {
box-sizing: border-box;
padding: 0;
}
table {
border-collapse: collapse;
border-spacing: 0;
}
td,
th {
padding: 0;
}
* {
box-sizing: border-box;
}
input {
font: 13px / 1.4 Helvetica, arial, nimbussansl, liberationsans, freesans, clean, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
a {
color: #4078c0;
text-decoration: none;
}
a:hover,
a:active {
text-decoration: underline;
}
hr {
height: 0;
margin: 15px 0;
overflow: hidden;
background: transparent;
border: 0;
border-bottom: 1px solid #ddd;
}
hr:before {
display: table;
content: "";
}
hr:after {
display: table;
clear: both;
content: "";
}
h1,
h2,
h3,
h4,
h5,
h6 {
margin-top: 15px;
margin-bottom: 15px;
line-height: 1.1;
}
h1 {
font-size: 30px;
}
h2 {
font-size: 21px;
}
h3 {
font-size: 16px;
}
h4 {
font-size: 14px;
}
h5 {
font-size: 12px;
}
h6 {
font-size: 11px;
}
blockquote {
margin: 0;
}
ul,
ol {
padding: 0;
margin-top: 0;
margin-bottom: 0;
}
ol ol,
ul ol {
list-style-type: lower-roman;
}
ul ul ol,
ul ol ol,
ol ul ol,
ol ol ol {
list-style-type: lower-alpha;
}
dd {
margin-left: 0;
}
code {
font-family: Consolas, "Liberation Mono", Menlo, Courier, monospace;
font-size: 12px;
}
pre {
margin-top: 0;
margin-bottom: 0;
font: 12px Consolas, "Liberation Mono", Menlo, Courier, monospace;
}
.select::-ms-expand {
opacity: 0;
}
.octicon {
font: normal normal normal 16px/1 octicons-link;
display: inline-block;
text-decoration: none;
text-rendering: auto;
-webkit-font-smoothing: antialiased;
-moz-osx-font-smoothing: grayscale;
-webkit-user-select: none;
-moz-user-select: none;
-ms-user-select: none;
user-select: none;
}
.octicon-link:before {
content: '\f05c';
}
.markdown-body:before {
display: table;
content: "";
}
.markdown-body:after {
display: table;
clear: both;
content: "";
}
.markdown-body>*:first-child {
margin-top: 0 !important;
}
.markdown-body>*:last-child {
margin-bottom: 0 !important;
}
a:not([href]) {
color: inherit;
text-decoration: none;
}
.anchor {
display: inline-block;
padding-right: 2px;
margin-left: -18px;
}
.anchor:focus {
outline: none;
}
h1,
h2,
h3,
h4,
h5,
h6 {
margin-top: 1em;
margin-bottom: 16px;
font-weight: bold;
line-height: 1.4;
}
h1 .octicon-link,
h2 .octicon-link,
h3 .octicon-link,
h4 .octicon-link,
h5 .octicon-link,
h6 .octicon-link {
color: #000;
vertical-align: middle;
visibility: hidden;
}
h1:hover .anchor,
h2:hover .anchor,
h3:hover .anchor,
h4:hover .anchor,
h5:hover .anchor,
h6:hover .anchor {
text-decoration: none;
}
h1:hover .anchor .octicon-link,
h2:hover .anchor .octicon-link,
h3:hover .anchor .octicon-link,
h4:hover .anchor .octicon-link,
h5:hover .anchor .octicon-link,
h6:hover .anchor .octicon-link {
visibility: visible;
}
h1 {
padding-bottom: 0.3em;
font-size: 2.25em;
line-height: 1.2;
border-bottom: 1px solid #eee;
}
h1 .anchor {
line-height: 1;
}
h2 {
padding-bottom: 0.3em;
font-size: 1.75em;
line-height: 1.225;
border-bottom: 1px solid #eee;
}
h2 .anchor {
line-height: 1;
}
h3 {
font-size: 1.5em;
line-height: 1.43;
}
h3 .anchor {
line-height: 1.2;
}
h4 {
font-size: 1.25em;
}
h4 .anchor {
line-height: 1.2;
}
h5 {
font-size: 1em;
}
h5 .anchor {
line-height: 1.1;
}
h6 {
font-size: 1em;
color: #777;
}
h6 .anchor {
line-height: 1.1;
}
p,
blockquote,
ul,
ol,
dl,
table,
pre {
margin-top: 0;
margin-bottom: 16px;
}
hr {
height: 4px;
padding: 0;
margin: 16px 0;
background-color: #e7e7e7;
border: 0 none;
}
ul,
ol {
padding-left: 2em;
}
ul ul,
ul ol,
ol ol,
ol ul {
margin-top: 0;
margin-bottom: 0;
}
li>p {
margin-top: 16px;
}
dl {
padding: 0;
}
dl dt {
padding: 0;
margin-top: 16px;
font-size: 1em;
font-style: italic;
font-weight: bold;
}
dl dd {
padding: 0 16px;
margin-bottom: 16px;
}
blockquote {
padding: 0 15px;
color: #777;
border-left: 4px solid #ddd;
}
blockquote>:first-child {
margin-top: 0;
}
blockquote>:last-child {
margin-bottom: 0;
}
table {
display: block;
width: 100%;
overflow: auto;
word-break: normal;
word-break: keep-all;
}
table th {
font-weight: bold;
}
table th,
table td {
padding: 6px 13px;
border: 1px solid #ddd;
}
table tr {
background-color: #fff;
border-top: 1px solid #ccc;
}
table tr:nth-child(2n) {
background-color: #f8f8f8;
}
img {
max-width: 100%;
box-sizing: content-box;
background-color: #fff;
}
code {
padding: 0;
padding-top: 0.2em;
padding-bottom: 0.2em;
margin: 0;
font-size: 85%;
background-color: rgba(0,0,0,0.04);
border-radius: 3px;
}
code:before,
code:after {
letter-spacing: -0.2em;
content: "\00a0";
}
pre>code {
padding: 0;
margin: 0;
font-size: 100%;
word-break: normal;
white-space: pre;
background: transparent;
border: 0;
}
.highlight {
margin-bottom: 16px;
}
.highlight pre,
pre {
padding: 16px;
overflow: auto;
font-size: 85%;
line-height: 1.45;
background-color: #f7f7f7;
border-radius: 3px;
}
.highlight pre {
margin-bottom: 0;
word-break: normal;
}
pre {
word-wrap: normal;
}
pre code {
display: inline;
max-width: initial;
padding: 0;
margin: 0;
overflow: initial;
line-height: inherit;
word-wrap: normal;
background-color: transparent;
border: 0;
}
pre code:before,
pre code:after {
content: normal;
}
kbd {
display: inline-block;
padding: 3px 5px;
font-size: 11px;
line-height: 10px;
color: #555;
vertical-align: middle;
background-color: #fcfcfc;
border: solid 1px #ccc;
border-bottom-color: #bbb;
border-radius: 3px;
box-shadow: inset 0 -1px 0 #bbb;
}
.pl-c {
color: #969896;
}
.pl-c1,
.pl-s .pl-v {
color: #0086b3;
}
.pl-e,
.pl-en {
color: #795da3;
}
.pl-s .pl-s1,
.pl-smi {
color: #333;
}
.pl-ent {
color: #63a35c;
}
.pl-k {
color: #a71d5d;
}
.pl-pds,
.pl-s,
.pl-s .pl-pse .pl-s1,
.pl-sr,
.pl-sr .pl-cce,
.pl-sr .pl-sra,
.pl-sr .pl-sre {
color: #183691;
}
.pl-v {
color: #ed6a43;
}
.pl-id {
color: #b52a1d;
}
.pl-ii {
background-color: #b52a1d;
color: #f8f8f8;
}
.pl-sr .pl-cce {
color: #63a35c;
font-weight: bold;
}
.pl-ml {
color: #693a17;
}
.pl-mh,
.pl-mh .pl-en,
.pl-ms {
color: #1d3e81;
font-weight: bold;
}
.pl-mq {
color: #008080;
}
.pl-mi {
color: #333;
font-style: italic;
}
.pl-mb {
color: #333;
font-weight: bold;
}
.pl-md {
background-color: #ffecec;
color: #bd2c00;
}
.pl-mi1 {
background-color: #eaffea;
color: #55a532;
}
.pl-mdr {
color: #795da3;
font-weight: bold;
}
.pl-mo {
color: #1d3e81;
}
kbd {
display: inline-block;
padding: 3px 5px;
font: 11px Consolas, "Liberation Mono", Menlo, Courier, monospace;
line-height: 10px;
color: #555;
vertical-align: middle;
background-color: #fcfcfc;
border: solid 1px #ccc;
border-bottom-color: #bbb;
border-radius: 3px;
box-shadow: inset 0 -1px 0 #bbb;
}
.task-list-item {
list-style-type: none;
}
.task-list-item+.task-list-item {
margin-top: 3px;
}
.task-list-item input {
margin: 0 0.35em 0.25em -1.6em;
vertical-align: middle;
}
:checked+.radio-label {
z-index: 1;
position: relative;
border-color: #4078c0;
}
.sourceLine {
display: inline-block;
}
code .kw { color: #000000; }
code .dt { color: #ed6a43; }
code .dv { color: #009999; }
code .bn { color: #009999; }
code .fl { color: #009999; }
code .ch { color: #009999; }
code .st { color: #183691; }
code .co { color: #969896; }
code .ot { color: #0086b3; }
code .al { color: #a61717; }
code .fu { color: #63a35c; }
code .er { color: #a61717; background-color: #e3d2d2; }
code .wa { color: #000000; }
code .cn { color: #008080; }
code .sc { color: #008080; }
code .vs { color: #183691; }
code .ss { color: #183691; }
code .im { color: #000000; }
code .va {color: #008080; }
code .cf { color: #000000; }
code .op { color: #000000; }
code .bu { color: #000000; }
code .ex { color: #000000; }
code .pp { color: #999999; }
code .at { color: #008080; }
code .do { color: #969896; }
code .an { color: #008080; }
code .cv { color: #008080; }
code .in { color: #008080; }
</style>
<style>
body {
  box-sizing: border-box;
  min-width: 200px;
  max-width: 980px;
  margin: 0 auto;
  padding: 45px;
  padding-top: 0px;
}
</style>


</head>

<body>

<h1 id="run-cox-proportional-hazards-regression-on-one-continuous-variable-with-quantile-cutoff">Run
Cox proportional hazards regression on one continuous variable with
quantile cutoff</h1>
<ul>
<li><a href="#using-single-continous-variable" id="toc-using-single-continous-variable">Using single continous
variable</a>
<ul>
<li><a href="#user-defined-quantile-cutoffs-to-group-samples-run_coxph_1d" id="toc-user-defined-quantile-cutoffs-to-group-samples-run_coxph_1d">User
defined quantile cutoff(s) to group samples
<code>run_coxph_1d</code></a>
<ul>
<li><a href="#with-one-cutoff" id="toc-with-one-cutoff">With one
cutoff</a></li>
<li><a href="#with-both-lower-and-upper-quantile-cutoff" id="toc-with-both-lower-and-upper-quantile-cutoff">With both lower and
upper quantile cutoff</a></li>
</ul></li>
<li><a href="#scanning-a-gradient-of-quantile-cutoffs-to-find-optimal-cutoff" id="toc-scanning-a-gradient-of-quantile-cutoffs-to-find-optimal-cutoff">Scanning
a gradient of quantile cutoff(s) to find optimal cutoff</a>
<ul>
<li><a href="#with-one-cutoff-1" id="toc-with-one-cutoff-1">With one
cutoff</a></li>
<li><a href="#with-both-lower-and-upper-quantile-cutoff-1" id="toc-with-both-lower-and-upper-quantile-cutoff-1">With both lower and
upper quantile cutoff</a></li>
</ul></li>
</ul></li>
<li><a href="#using-two-continous-variable" id="toc-using-two-continous-variable">Using two continous variable</a>
<ul>
<li><a href="#user-defined-quantile-cutoffs-to-group-samples-run_coxph_2d" id="toc-user-defined-quantile-cutoffs-to-group-samples-run_coxph_2d">User
defined quantile cutoff(s) to group samples
<code>run_coxph_2d</code></a>
<ul>
<li><a href="#with-one-cutoff-2" id="toc-with-one-cutoff-2">With one
cutoff</a></li>
<li><a href="#with-both-lower-and-upper-quantile-cutoff-2" id="toc-with-both-lower-and-upper-quantile-cutoff-2">With both lower and
upper quantile cutoff</a></li>
</ul></li>
<li><a href="#scanning-a-gradient-of-quantile-cutoffs-to-find-optimal-cutoff-1" id="toc-scanning-a-gradient-of-quantile-cutoffs-to-find-optimal-cutoff-1">Scanning
a gradient of quantile cutoff(s) to find optimal cutoff</a>
<ul>
<li><a href="#with-one-cutoff-3" id="toc-with-one-cutoff-3">With one
cutoff</a></li>
<li><a href="#with-both-lower-and-upper-quantile-cutoff-3" id="toc-with-both-lower-and-upper-quantile-cutoff-3">With both lower and
upper quantile cutoff</a></li>
</ul></li>
</ul></li>
<li><a href="#extract-tcga-clinical-and-omics-data-from-cbioportal" id="toc-extract-tcga-clinical-and-omics-data-from-cbioportal">Extract
TCGA clinical and omics data from cBioPortal</a></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cutsurv)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># library(skimr)</span></span></code></pre></div>
<p>We will use HNSC data from the package for the examples below. To
know more about HNSC data, refer to <code>?HNSC</code></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(HNSC)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>HNSC</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 502 × 23</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    sample_id   subje…¹ tumor…² sampl…³ age_a…⁴ gender race  SURVI…⁵ SURVI…⁶ SURVI…⁷</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 TCGA-4P-AA… TCGA-4… HNSC    Primar…      66 male   Blac…     102       0     102</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 TCGA-BA-40… TCGA-B… HNSC    Primar…      69 male   White     462       1     396</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 TCGA-BA-40… TCGA-B… HNSC    Primar…      49 male   Blac…     283       1     236</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 TCGA-BA-40… TCGA-B… HNSC    Primar…      39 male   White     415       1     286</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 TCGA-BA-40… TCGA-B… HNSC    Primar…      45 female White    1134       1    1134</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 TCGA-BA-40… TCGA-B… HNSC    Primar…      83 male   White     276       1     276</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 TCGA-BA-51… TCGA-B… HNSC    Primar…      72 male   White     722       0     517</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 TCGA-BA-51… TCGA-B… HNSC    Primar…      56 male   White    1288       0    1288</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 TCGA-BA-51… TCGA-B… HNSC    Primar…      51 male   White    1762       1    1522</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 TCGA-BA-55… TCGA-B… HNSC    Primar…      54 male   Blac…     520       0     248</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 492 more rows, 13 more variables: SURVIVAL_pfi_event &lt;dbl&gt;,</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   cancer_grading &lt;chr&gt;, tnm_stage_pathologic_overall_stage &lt;chr&gt;,</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   TNFRSF1B &lt;dbl&gt;, `t_nk_cell:CD4+ T` &lt;dbl&gt;, `t_nk_cell:CD8+ T` &lt;dbl&gt;,</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `t_nk_cell:Treg` &lt;dbl&gt;, `t_nk_cell:Cytotoxic` &lt;dbl&gt;, `t_nk_cell:NK` &lt;dbl&gt;,</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `t_nk_cell:Exhausted` &lt;dbl&gt;, `myeloid:Anti-inflammatory` &lt;dbl&gt;,</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `myeloid:Pro-inflammatory` &lt;dbl&gt;, subtype &lt;chr&gt;, and abbreviated variable</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   names ¹​subject_id, ²​tumor_type, ³​sample_type, ⁴​age_at_diagnosis_years, …</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co"># skimr::skim(HNSC)</span></span></code></pre></div>
<h1 id="using-single-continous-variable">Using single continous
variable</h1>
<h2 id="user-defined-quantile-cutoffs-to-group-samples-run_coxph_1d">User
defined quantile cutoff(s) to group samples
<code>run_coxph_1d</code></h2>
<p>Here we use overall survival response to regress against the
expression of <em>TNFRSF1B</em>, stratified by patient cancer
grading.</p>
<h3 id="with-one-cutoff">With one cutoff</h3>
<p>Discretising patients by the median expression of <em>TNFRSF1B</em>,
with higher than median group as baseline.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_1d</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B <span class="sc">+</span> <span class="fu">strata</span>(cancer_grading), <span class="at">input_data =</span> HNSC, <span class="at">q_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">q =</span> <span class="fl">0.5</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;test&quot;</span>]]</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; survival::coxph(formula = new_formula, data = tmp)</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            coef exp(coef) se(coef)     z      p</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow 0.3168    1.3728   0.1401 2.262 0.0237</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihood ratio test=5.17  on 1 df, p=0.02296</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; n= 498, number of events= 217</span></span></code></pre></div>
<p>Result also returns discretized input data which contains all the
variables in formula and can be used for other model fit in the
future.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;data&quot;</span>]]</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 498 × 5</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    SURVIVAL_os_days SURVIVAL_os_event TNFRSF1B cancer_grading group</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;               &lt;dbl&gt;             &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;          &lt;fct&gt;</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1              102                 0     3.61 G2             high </span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2              462                 1     1.68 G3             low  </span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3              283                 1     1.87 G2             low  </span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4              415                 1     1.81 G2             low  </span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5             1134                 1     3.88 G2             high </span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6              276                 1     2.74 G2             low  </span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7              722                 0     3.51 G1             high </span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8             1288                 0     4.41 G2             high </span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9             1762                 1     4.74 G2             high </span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10              520                 0     3.17 G2             low  </span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 488 more rows</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ Use `print(n = ...)` to see more rows</span></span></code></pre></div>
<h3 id="with-both-lower-and-upper-quantile-cutoff">With both lower and
upper quantile cutoff</h3>
<ul>
<li><p>Expression of <em>TNFRSF1B</em> can also be binned to lower than
quantile <code>q</code> group and higher than quantile <code>q_2</code>
group.</p></li>
<li><p>We can also switch the baseline for regression. The default from
above is “high”.</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_1d</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B <span class="sc">+</span> <span class="fu">strata</span>(cancer_grading), <span class="at">input_data =</span> HNSC, <span class="at">q_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">q =</span> <span class="fl">0.4</span>, <span class="at">q_2=</span><span class="fl">0.6</span>, <span class="at">baseline =</span> <span class="st">&quot;low&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;test&quot;</span>]]</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; survival::coxph(formula = new_formula, data = tmp)</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;              coef exp(coef) se(coef)      z     p</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouphigh -0.3268    0.7212   0.1550 -2.109 0.035</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihood ratio test=4.48  on 1 df, p=0.03423</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; n= 398, number of events= 178</span></span></code></pre></div>
<p>For more helps, <code>?run_coxph_1d</code>.</p>
<h2 id="scanning-a-gradient-of-quantile-cutoffs-to-find-optimal-cutoff">Scanning
a gradient of quantile cutoff(s) to find optimal cutoff</h2>
<h3 id="with-one-cutoff-1">With one cutoff</h3>
<p><code>run_coxph_1d_optimal</code> performs Cox proportional hazards
regression with multiple quantile cutoff values and returns the best
test result, best data, best quantile cutoff, and a dataframe of the
quantile cutoffs ranked by the selected p-value type.</p>
<ul>
<li>The cutoff gradient can be defined by <code>low_q</code>,
<code>high_q</code> and <code>breaks</code> or <code>by</code>, very
similarly to <code>{seq}</code> function.</li>
<li>The best cutoff are selected by choosing threshold with the lowest
model p-value. There are several types of p-value for ranking.
<code>&quot;log&quot;</code> means likelihood ratio p-value, <code>&quot;sc&quot;</code>
means log rank score test p-value, and <code>&quot;wald&quot;</code> means wald
test p-value. Default is <code>&quot;log&quot;</code></li>
<li>After discretion, the groups with lower than
<code>min_group_size</code> will be removed from regression.</li>
<li>Like <code>run_coxph_1d</code>, we can also choose
<code>baseline</code> for the group level.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_1d_optimal</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B, <span class="at">input_data =</span> HNSC, <span class="at">q_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">by =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<p>The result include 4 parts:</p>
<ul>
<li>“best_test”: The coxph test object with the lowest p-value.</li>
<li>“best_data”: The data frame used to fit the best test.</li>
<li>“best_q”: The quantile cutoff value that produced the best
test.</li>
<li>“rank_q”: A data frame with the ranked quantile cutoff values and
their corresponding p-values for the selected p-value type.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;best_test&quot;</span>]]</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; survival::coxph(formula = new_formula, data = tmp)</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            coef exp(coef) se(coef)     z      p</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow 0.3603    1.4338   0.1426 2.528 0.0115</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihood ratio test=6.16  on 1 df, p=0.01306</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; n= 501, number of events= 218</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;best_q&quot;</span>]]</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   q </span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.3</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;rank_q&quot;</span>]]</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 7 × 3</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       q cutoff p.value.log</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;dbl&gt;  &lt;dbl&gt;       &lt;dbl&gt;</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1   0.3   2.69      0.0131</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2   0.4   2.96      0.0136</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3   0.5   3.25      0.0146</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4   0.2   2.31      0.0633</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5   0.7   3.93      0.0771</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6   0.8   4.21      0.0895</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7   0.6   3.54      0.108</span></span></code></pre></div>
<p>To learn more, <code>?run_coxph_1d_optimal</code>.</p>
<h3 id="with-both-lower-and-upper-quantile-cutoff-1">With both lower and
upper quantile cutoff</h3>
<p><code>run_coxph_1d2b_optimal</code> is similar to
<code>run_coxph_1d_optimal</code> but with both lower <code>q</code> and
upper <code>q_2</code> quantile cutoffs to group samples. All other
parameters are the same with 2 additional parameters.</p>
<ul>
<li>Method for generating the search space for optimal cutoff values.
Inspired from cSurvival paper<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, there are two methods included:
“median-anchored” and “exhaustive”.</li>
</ul>
<blockquote>
<ol style="list-style-type: decimal">
<li>Median-anchored greedy search: we construct a 2D grid using
percentiles of both predictors. Next, we determine the starting point
for a greedy search by locating the minimum P-value computed from
testing each percentile in predictor B against the median percentile in
predictor A. Then, we test the nearest three unexplored points; if a
lower P-value is found, we move the search to that newly found minimum
P-value point and test the nearest unexplored points until no lower
P-value can be found.</li>
</ol>
</blockquote>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>Exhaustive search: We construct a 2D grid using percentiles of both
predictors. Next, we determine the optimal percentile combination by
locating the minimum P-value computed from testing each percentile in
predictor B against each percentile in predictor A.</li>
</ol>
</blockquote>
<ul>
<li>set_n: The number of quantitle combinations closest to the previous
set of cutoff generating lowest p-value for the median-anchored method.
Default is 3.</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_1d2b_optimal</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B, <span class="at">input_data =</span> HNSC, <span class="at">q_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">low_q =</span> <span class="fl">0.2</span>, <span class="at">high_q =</span> <span class="fl">0.8</span>,<span class="at">by =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;best_test&quot;</span>]]</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; survival::coxph(formula = new_formula, data = tmp)</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            coef exp(coef) se(coef)    z       p</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow 0.4325    1.5412   0.1539 2.81 0.00495</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihood ratio test=7.74  on 1 df, p=0.005398</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; n= 402, number of events= 175</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;best_q&quot;</span>]]</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   q q_2 </span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.3 0.5</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;rank_q&quot;</span>]]</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 16 × 5</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        q   q_2 cutoff cutoff_2 p.value.log</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;       &lt;dbl&gt;</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1   0.3   0.5   2.51     3.52     0.00540</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2   0.4   0.5   2.86     3.38     0.00785</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3   0.3   0.4   2.61     3.11     0.00962</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4   0.3   0.6   2.37     4.00     0.0130 </span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5   0.3   0.3   2.69     2.69     0.0131 </span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6   0.4   0.4   2.96     2.96     0.0136 </span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7   0.5   0.5   3.25     3.25     0.0146 </span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8   0.2   0.5   2.01     3.71     0.0182 </span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9   0.4   0.6   2.72     3.85     0.0200 </span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10   0.2   0.4   2.10     3.29     0.0299 </span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11   0.2   0.6   1.91     4.11     0.0331 </span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12   0.2   0.3   2.23     2.86     0.0377 </span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13   0.2   0.2   2.31     2.31     0.0633 </span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14   0.7   0.7   3.93     3.93     0.0771 </span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15   0.8   0.8   4.21     4.21     0.0895 </span></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16   0.6   0.6   3.54     3.54     0.108</span></span></code></pre></div>
<p>For more information, <code>?run_coxph_1d2b_optimal</code>.</p>
<h1 id="using-two-continous-variable">Using two continous variable</h1>
<h2 id="user-defined-quantile-cutoffs-to-group-samples-run_coxph_2d">User
defined quantile cutoff(s) to group samples
<code>run_coxph_2d</code></h2>
<p>Here we use overall survival response to regress against the
expression of <em>TNFRSF1B</em> and the GSVA enrichment score of Treg
<code>&quot;t_nk_cell:Treg&quot;</code>, stratified by patient cancer grading and
gender.</p>
<h3 id="with-one-cutoff-2">With one cutoff</h3>
<p>Discretising patients by the median expression of <em>TNFRSF1B</em>
and median enrichment score, with higher than median group as baseline.
It will group samples into 4 groups: “high:high”, “high:low”,
“low:high”, “low:low”.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_2d</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B <span class="sc">+</span> <span class="st">`</span><span class="at">t_nk_cell:Treg</span><span class="st">`</span> <span class="sc">+</span> <span class="fu">strata</span>(cancer_grading, gender), <span class="at">input_data =</span> HNSC, <span class="at">q1_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">q2_col =</span> <span class="st">&quot;t_nk_cell:Treg&quot;</span>, <span class="at">q1=</span><span class="fl">0.5</span>, <span class="at">q2 =</span> <span class="fl">0.5</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;test&quot;</span>]]</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; survival::coxph(formula = new_formula, data = tmp)</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                  coef exp(coef) se(coef)      z      p</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouphigh:low -0.4016    0.6693   0.2675 -1.501 0.1333</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow:high  0.0312    1.0317   0.2129  0.147 0.8835</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow:low   0.3029    1.3538   0.1608  1.883 0.0597</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihood ratio test=9.21  on 3 df, p=0.02667</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; n= 498, number of events= 217</span></span></code></pre></div>
<p>Same as <code>run_coxph_1d</code>, result also returns discretized
input data which contains all the variables in formula and can be used
for other model fit in the future.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;data&quot;</span>]]</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 498 × 7</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    SURVIVAL_os_days SURVIVAL_os_event TNFRSF1B t_nk_cell:Tre…¹ cance…² gender group</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;               &lt;dbl&gt;             &lt;dbl&gt;    &lt;dbl&gt;           &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;  &lt;fct&gt;</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1              102                 0     3.61           0.665 G2      male   high…</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2              462                 1     1.68           0.540 G3      male   low:…</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3              283                 1     1.87           0.274 G2      male   low:…</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4              415                 1     1.81           0.596 G2      male   low:…</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5             1134                 1     3.88           0.803 G2      female high…</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6              276                 1     2.74           0.641 G2      male   low:…</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7              722                 0     3.51           0.666 G1      male   high…</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8             1288                 0     4.41           0.760 G2      male   high…</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9             1762                 1     4.74           0.554 G2      male   high…</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10              520                 0     3.17           0.607 G2      male   low:…</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 488 more rows, and abbreviated variable names ¹​`t_nk_cell:Treg`,</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ²​cancer_grading</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ Use `print(n = ...)` to see more rows</span></span></code></pre></div>
<p>We can also combine some groups into one, so that instead of
regressing against 4 groups, we only interested in whether either TNFR2
low and Treg high enrichment will result in poor prognosis. Thus,
“high:low”, “low:low”, “high:high” will be integrated into one group
<code>&quot;other&quot;</code>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_2d</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B <span class="sc">+</span> <span class="st">`</span><span class="at">t_nk_cell:Treg</span><span class="st">`</span> <span class="sc">+</span> <span class="fu">strata</span>(cancer_grading, gender), <span class="at">input_data =</span> HNSC, <span class="at">q1_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">q2_col =</span> <span class="st">&quot;t_nk_cell:Treg&quot;</span>, <span class="at">q1=</span><span class="fl">0.5</span>, <span class="at">q2 =</span> <span class="fl">0.5</span>, <span class="at">combine_group =</span> <span class="fu">c</span>(<span class="st">&quot;high:low&quot;</span>, <span class="st">&quot;low:low&quot;</span>, <span class="st">&quot;high:high&quot;</span>))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;test&quot;</span>]]</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; survival::coxph(formula = new_formula, data = tmp)</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                  coef exp(coef) se(coef)     z    p</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow:high -0.0542    0.9472   0.1937 -0.28 0.78</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihood ratio test=0.08  on 1 df, p=0.7783</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; n= 498, number of events= 217</span></span></code></pre></div>
<h3 id="with-both-lower-and-upper-quantile-cutoff-2">With both lower and
upper quantile cutoff</h3>
<p>Same as <code>run_coxph_1d</code>, both variables can also be binned
to lower than quantile <code>q1</code> and/or <code>q2</code> and higher
than quantile <code>q1_2</code> and/or <code>q2_2</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_2d</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B <span class="sc">+</span> <span class="st">`</span><span class="at">t_nk_cell:Treg</span><span class="st">`</span> <span class="sc">+</span> <span class="fu">strata</span>(cancer_grading, gender), <span class="at">input_data =</span> HNSC, <span class="at">q1_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">q2_col =</span> <span class="st">&quot;t_nk_cell:Treg&quot;</span>, <span class="at">q1=</span><span class="fl">0.4</span>, <span class="at">q1_2 =</span> <span class="fl">0.6</span>, <span class="at">q2 =</span> <span class="fl">0.5</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;test&quot;</span>]]</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; survival::coxph(formula = new_formula, data = tmp)</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                  coef exp(coef) se(coef)      z      p</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouphigh:low -0.5270    0.5904   0.3201 -1.646 0.0997</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow:high  0.1124    1.1189   0.2393  0.470 0.6387</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow:low   0.2602    1.2971   0.1773  1.467 0.1424</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihood ratio test=7.81  on 3 df, p=0.05019</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; n= 398, number of events= 178</span></span></code></pre></div>
<p>For more helps, <code>?run_coxph_2d</code>.</p>
<h2 id="scanning-a-gradient-of-quantile-cutoffs-to-find-optimal-cutoff-1">Scanning
a gradient of quantile cutoff(s) to find optimal cutoff</h2>
<p><code>run_coxph_2d_optimal</code> and
<code>run_coxph_2d2d_optimal</code> perform coxph survival analysis by
converting two continuous variables to categorical variables with
optimal quantile cutoffs. The optimal cutoffs are determined by
exhaustively searching or using median-anchored method. The optimal
cutoffs are determined by minimizing the p-value of the score test of
the Cox proportional hazards model.</p>
<h3 id="with-one-cutoff-3">With one cutoff</h3>
<p><code>run_coxph_2d_optimal</code> is to search combination of
<code>q1</code> and <code>q2</code> for both variables. <code>q1</code>
and <code>q2</code> separate var1 and var2 into two groups respectively.
It is flexible with all the parameters defined in
<code>run_coxph_1d_optimal</code> seen above, but search 2-dimensional
space.</p>
<p>Below is the example to evaluate, comparing to both TNFR2 and Treg
enrichment high, whether single high or both low result in poor or
better prognosis using “median-anchored” to search the optimal pair of
<code>q1</code> and <code>q2</code> cutoff.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_2d_optimal</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B <span class="sc">+</span> <span class="st">`</span><span class="at">t_nk_cell:Treg</span><span class="st">`</span> <span class="sc">+</span> <span class="fu">strata</span>(cancer_grading), <span class="at">input_data =</span> HNSC, <span class="at">q1_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">q2_col =</span> <span class="st">&quot;t_nk_cell:Treg&quot;</span>, <span class="at">by=</span><span class="fl">0.1</span>, <span class="at">combine_group =</span> <span class="fu">c</span>(<span class="st">&quot;low:high&quot;</span>, <span class="st">&quot;high:low&quot;</span>), <span class="at">baseline=</span><span class="st">&quot;high:high&quot;</span>, <span class="at">set_n =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;best_test&quot;</span>]]</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; survival::coxph(formula = new_formula, data = tmp)</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                 coef exp(coef) se(coef)      z      p</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; grouplow:low  0.2874    1.3330   0.1591  1.807 0.0708</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; groupother   -0.1615    0.8509   0.1835 -0.880 0.3789</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihood ratio test=7.03  on 2 df, p=0.02968</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; n= 498, number of events= 217</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;best_q&quot;</span>]]</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  q1  q2 </span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.5 0.5</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>res[[<span class="st">&quot;rank_q&quot;</span>]]</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 12 × 5</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       q1 cutoff1    q2 cutoff2 p.value.log</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;       &lt;dbl&gt;</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1   0.5    3.25   0.5   0.630      0.0297</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2   0.4    2.96   0.6   0.668      0.0312</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3   0.6    3.54   0.5   0.630      0.0319</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4   0.5    3.25   0.6   0.668      0.0399</span></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5   0.4    2.96   0.5   0.630      0.0506</span></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6   0.3    2.70   0.5   0.630      0.0655</span></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7   0.4    2.96   0.4   0.581      0.0950</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8   0.2    2.34   0.5   0.630      0.100 </span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9   0.5    3.25   0.4   0.581      0.109 </span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10   0.6    3.54   0.4   0.581      0.164 </span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11   0.7    3.93   0.5   0.630      0.209 </span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12   0.8    4.21   0.5   0.630      0.302</span></span></code></pre></div>
<p>For more information, <code>?run_coxph_2d_optimal</code></p>
<h3 id="with-both-lower-and-upper-quantile-cutoff-3">With both lower and
upper quantile cutoff</h3>
<p>Same as <code>run_coxph_1d2b_optimal</code>,
<code>run_coxph_2d2b_optimal</code> search the combination of
<code>q1</code>, <code>q2</code>, <code>q1_2</code> and
<code>q2_2</code>, binning samples by lower and upper quantile
cutoff.</p>
<p>Below is the example same as above, but with lower and upper quantile
cutoffs.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_coxph_2d2b_optimal</span>(<span class="at">formula=</span><span class="fu">Surv</span>(SURVIVAL_os_days, SURVIVAL_os_event) <span class="sc">~</span> TNFRSF1B <span class="sc">+</span> <span class="st">`</span><span class="at">t_nk_cell:Treg</span><span class="st">`</span> <span class="sc">+</span> <span class="fu">strata</span>(cancer_grading), <span class="at">input_data =</span> HNSC, <span class="at">q1_col=</span><span class="st">&quot;TNFRSF1B&quot;</span>, <span class="at">q2_col =</span> <span class="st">&quot;t_nk_cell:Treg&quot;</span>, <span class="at">by=</span><span class="fl">0.1</span>, <span class="at">combine_group =</span> <span class="fu">c</span>(<span class="st">&quot;low:high&quot;</span>, <span class="st">&quot;high:low&quot;</span>), <span class="at">baseline=</span><span class="st">&quot;high:high&quot;</span>, <span class="at">set_n =</span> <span class="dv">5</span>)</span></code></pre></div>
<h1 id="extract-tcga-clinical-and-omics-data-from-cbioportal">Extract
TCGA clinical and omics data from cBioPortal</h1>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_tcga</span>(<span class="at">tumor_type=</span><span class="st">&quot;BRCA&quot;</span>) <span class="co"># get clinical</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1,108 × 8</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    patient_id   sample_type sample_id       os_months os_status  age   race  sex   </span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;        &lt;chr&gt;       &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; </span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 TCGA-AR-A1AR Primary     TCGA-AR-A1AR-01 17.21     1:DECEASED 50    WHITE Female</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 TCGA-BH-A1EO Primary     TCGA-BH-A1EO-01 91.92     1:DECEASED 68    WHITE Female</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 TCGA-BH-A1ES Primary     TCGA-BH-A1ES-01 113.73    1:DECEASED 35    WHITE Female</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 TCGA-BH-A1ES Metastasis  TCGA-BH-A1ES-06 113.73    1:DECEASED 35    WHITE Female</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 TCGA-BH-A1ET Primary     TCGA-BH-A1ET-01 82.79     1:DECEASED 55    WHITE Female</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 TCGA-BH-A1EU Primary     TCGA-BH-A1EU-01 42.25     1:DECEASED 83    WHITE Female</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 TCGA-BH-A1EV Primary     TCGA-BH-A1EV-01 11.99     1:DECEASED 45    WHITE Female</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 TCGA-BH-A1EW Primary     TCGA-BH-A1EW-01 55.65     1:DECEASED 38    WHITE Female</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 TCGA-BH-A1F0 Primary     TCGA-BH-A1F0-01 25.79     1:DECEASED 80    WHITE Female</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 TCGA-C8-A1HF Primary     TCGA-C8-A1HF-01 10.91     0:LIVING   48    ASIAN Female</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 1,098 more rows</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ Use `print(n = ...)` to see more rows</span></span></code></pre></div>
<p>Get gene expression for <em>BRCA1</em> and <em>BRCA2</em>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_tcga</span>(<span class="at">tumor_type=</span><span class="st">&quot;BRCA&quot;</span>, <span class="at">gene_names=</span><span class="fu">c</span>(<span class="st">&quot;BRCA1&quot;</span>, <span class="st">&quot;BRCA2&quot;</span>), <span class="at">type=</span><span class="st">&quot;expression&quot;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 2,200 × 10</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    uniqueSamp…¹ uniqu…² entre…³ molec…⁴ sampl…⁵ patie…⁶ studyId value hugoG…⁷ type </span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;        &lt;chr&gt;     &lt;int&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 VENHQS0zQy1… VENHQS…     672 brca_t… TCGA-3… TCGA-3… brca_t…  831. BRCA1   prot…</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 VENHQS0zQy1… VENHQS…     675 brca_t… TCGA-3… TCGA-3… brca_t…  179. BRCA2   prot…</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 VENHQS0zQy1… VENHQS…     672 brca_t… TCGA-3… TCGA-3… brca_t…  390. BRCA1   prot…</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 VENHQS0zQy1… VENHQS…     675 brca_t… TCGA-3… TCGA-3… brca_t…  154. BRCA2   prot…</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 VENHQS0zQy1… VENHQS…     672 brca_t… TCGA-3… TCGA-3… brca_t…  200. BRCA1   prot…</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 VENHQS0zQy1… VENHQS…     675 brca_t… TCGA-3… TCGA-3… brca_t…  151. BRCA2   prot…</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 VENHQS0zQy1… VENHQS…     672 brca_t… TCGA-3… TCGA-3… brca_t…  148. BRCA1   prot…</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 VENHQS0zQy1… VENHQS…     675 brca_t… TCGA-3… TCGA-3… brca_t…  102. BRCA2   prot…</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 VENHQS00SC1… VENHQS…     672 brca_t… TCGA-4… TCGA-4… brca_t…  376. BRCA1   prot…</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 VENHQS00SC1… VENHQS…     675 brca_t… TCGA-4… TCGA-4… brca_t…  129. BRCA2   prot…</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 2,190 more rows, and abbreviated variable names ¹​uniqueSampleKey,</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ²​uniquePatientKey, ³​entrezGeneId, ⁴​molecularProfileId, ⁵​sampleId, ⁶​patientId,</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ⁷​hugoGeneSymbol</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ Use `print(n = ...)` to see more rows</span></span></code></pre></div>
<p>For more information, <code>?get_tcga</code></p>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Xuanjin Cheng, Yongxing Liu, Jiahe Wang, Yujie Chen,
Andrew Gordon Robertson, Xuekui Zhang, Steven J M Jones, Stefan Taubert,
cSurvival: a web resource for biomarker interactions in cancer outcomes
and in cell lines, Briefings in Bioinformatics, Volume 23, Issue 3, May
2022, bbac090, <a href="https://doi.org/10.1093/bib/bbac090">https://doi.org/10.1093/bib/bbac090</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>

</body>
</html>
